Geometry/HGCalCommonData/plugins/DDHGCalTBModuleX.cc

```diff
@@ -0,0 +1,364 @@
+#ifdef EDM_ML_DEBUG
+    edm::LogVerbatim("HGCalGeom") << "DDHGCalTBModuleX: " << glog.name()
+				  << " number " << i << " positioned in "
+				  << parent.name() << " at " << r1 
+				  << " with " << rot;
+#endif
+    constructLayers(i,layerFrontIn_[i],layerBackIn_[i],-0.5*blockThick_[i],
+		    blockThick_[i],false,glog,cpv);
+    if (inOut_ > 1)
+      constructLayers(i,layerFrontOut_[i],layerBackOut_[i],-0.5*blockThick_[i],
+		      blockThick_[i],true,glog,cpv);
+    zi     = zo;
+  }
+#ifdef EDM_ML_DEBUG
+  edm::LogVerbatim("HGCalGeom") << "DDHGCalTBModuleX: All blocks are "
+				<< "placed in "	<< zMinBlock_ << ":" << zi;
+#endif
+}
+
+void DDHGCalTBModuleX::constructLayers(int block, int firstLayer, 
+				       int lastLayer, double zFront, 
+				       double totalWidth,  bool ignoreCenter,
+				       const DDLogicalPart& module, 
+				       DDCompactView& cpv) {
+  
+#ifdef EDM_ML_DEBUG
+  edm::LogVerbatim("HGCalGeom") << "DDHGCalTBModuleX: \t\tInside Block "
+				<< block << " Layers " << firstLayer << ":"
+				<< lastLayer << " zFront " << zFront
+				<< " thickness " << totalWidth
+				<< " ignore Center " << ignoreCenter;
+#endif
+  double       zi(zFront), thickTot(0);
+  for (int ly=firstLayer; ly<=lastLayer; ++ly) {
+    int     ii     = layerType_[ly];
+    int     copy   = copyNumber_[ii];
+    double  zz     = zi + (0.5*layerThick_[ii]);
+    double  zo     = zi + layerThick_[ii];
+    thickTot      += layerThick_[ii];
+
+    std::string name = "HGCal"+names_[ii]+std::to_string(copy);
+#ifdef EDM_ML_DEBUG
+    edm::LogVerbatim("HGCalGeom") << "DDHGCalTBModuleX: " << name << " Layer "
+				  << ly << ":" << ii << " Z " << zi << ":"
+				  << zo << " Thick " << layerThick_[ii]
+				  << " Sense " << layerSense_[ly];
+#endif
+    DDName matName(DDSplit(materials_[ii]).first, 
+		   DDSplit(materials_[ii]).second);
+    DDMaterial matter(matName);
+    DDLogicalPart glog;
+    if (layerSense_[ly] == 0) {
+      DDSolid solid = DDSolidFactory::box(DDName(name, idNameSpace_),
+					  absorbW_, absorbH_, 
+					  0.5*layerThick_[ii]);
+      glog = DDLogicalPart(solid.ddname(), matter, solid);
+#ifdef EDM_ML_DEBUG
+      edm::LogVerbatim("HGCalGeom") << "DDHGCalTBModuleX: " 
+				    << solid.name() << " box of dimension " 
+				    << absorbW_ << ":" << absorbH_
+				    << ":" << 0.5*layerThick_[ii];
+#endif
+      DDTranslation r1(0,0,zz);
+      DDRotation rot;
+      cpv.position(glog, module, copy, r1, rot);
+#ifdef EDM_ML_DEBUG
+      edm::LogVerbatim("HGCalGeom") << "DDHGCalTBModuleX: " << glog.name()
+				    << " number " << copy 
+				    << " positioned in " << module.name()
+				    << " at " << r1 << " with " << rot;
+#endif
+    } else if (layerSense_[ly] > 0) {
+      positionSensitive(zz, copy, layerSense_[ly], rMax_, maxModule_[ly],
+			ignoreCenter, module, cpv);
+    }
+    ++copyNumber_[ii];
+    zi     = zo;
+  } // End of loop over layers in a block
+
+  if (fabs(thickTot-totalWidth) < 0.00001) {
```

Comment:
Need to use a constant here:
`constexpr double tolerance = 0.00001;`
Maybe it should have a more specific name.

Reply:
Done